lib/limiter/clock.rb

```diff
@@ -0,0 +1,31 @@
+# frozen_string_literal: true
+
+require 'singleton'
+require 'forwardable'
+
+module Limiter
+  class Clock
+    include Singleton
+
+    extend SingleForwardable
+    def_single_delegators :instance, :skip, :time
+
+    def initialize
+      @offset = 0
+    end
+
+    def skip(interval)
```

Comment:
If I'm not mistaken `Clock#skip` is only used to advance the clock in tests à la Timecop, right? If so I'm wondering if `#skip` should be defined in `Clock` itself, because it's only relevant for testing. Do you think maybe it would be nicer to -if possible- stub `Clock#time` in the test instead?